### PR TITLE
tools: update our scripts with go{fmt,vet} 1.6

### PR DESCRIPTION
**Pull request by kargakis**
@stevekuznetsov @smarterclayton

### CODE REVIEW

**Comment by kargakis** [comment]
Removed from the tools repo at https://github.com/golang/tools/commit/adaaa074861b0d254acf51dec74daf7c2ba20e90

**Comment by mfojtik** [comment]
@kargakis ❤️

**Comment by mfojtik** [comment]
LGTM (fixed for source-to-image repo as well).

**Comment by soltysh** [comment]
LGTM

**Comment by stevekuznetsov** [comment]
Nice! LGTM

**Comment by liggitt** [comment]
@smarterclayton Are we still running these scripts using any versions of go that don't have go vet?

Also, will this expose Travis to differences between 1.5 and 1.6's govet?

**Comment by stevekuznetsov** [comment]
>will this expose Travis to differences between 1.5 and 1.6's govet

Good point, it will, although as far as I remember 1.6 is a superset of 1.5. We should only enable `hack/verify-govet.sh` for `go1.6`.

File: test/extended/builds/gitauth.go [hunk 1]
```diff
@@ -91,11 +91,10 @@ var _ = g.Describe("[builds][Slow] can use private repositories as build input",
 			testGitAuth(gitServerFixture, sourceURLTemplate, func() string {
 				g.By(fmt.Sprintf("creating a new secret for the gitserver by calling oc secrets new-basicauth %s --username=%s --password=%s --cacert=%s",
 					sourceSecretName, gitUserName, gitPassword, caCertPath))
-				err := oc.Run("secrets").
-					Args("new-basicauth", sourceSecretName,
-					fmt.Sprintf("--username=%s", gitUserName),
```
Comment (kargakis) [inline test/extended/builds/gitauth.go]: gofmt behaves differently between 1.5 and 1.6 here

**Comment by kargakis** [comment]
> @smarterclayton Are we still running these scripts using any versions of go that don't have go vet?

We switched Travis to use 1.5.3 and 1.6 - both of them include vet

Do we run these scripts in Jenkins? Why does Jenkins still builds using 1.4.2?

**Comment by smarterclayton** [comment]
Because RHEL and centos are still using go 1.4.

File: test/extended/builds/gitauth.go (hunk 1 above)
Comment (stevekuznetsov) [inline test/extended/builds/gitauth.go]: What warning resulted in these changes?

Comment (kargakis) [inline test/extended/builds/gitauth.go]: gofmt 1.6 uses one more tab to indent here - not a vet issue

**Comment by kargakis** [comment]
go vet in 1.6:
```
pkg/cmd/util/net.go:69: assignment copies lock value to *config: crypto/tls.Config contains sync.Once contains sync.Mutex
```
I don't want to remove those warnings but I am not sure how to repurpose that part of code[1]. Any ideas?
```go
func ListenAndServeTLS(srv *http.Server, network string, certFile, keyFile string) error {
	addr := srv.Addr
	if addr == "" {
		addr = ":https"
	}
	config := &tls.Config{}
	if srv.TLSConfig != nil {
[1]		*config = *srv.TLSConfig
	}
	if config.NextProtos == nil {
		config.NextProtos = []string{"http/1.1"}
	}

	var err error
	config.Certificates = make([]tls.Certificate, 1)
	config.Certificates[0], err = tls.LoadX509KeyPair(certFile, keyFile)
	if err != nil {
		return err
	}

	ln, err := net.Listen(network, addr)
	if err != nil {
		return err
	}

	tlsListener := tls.NewListener(tcpKeepAliveListener{ln.(*net.TCPListener)}, config)
	return srv.Serve(tlsListener)
}
```

**Comment by stevekuznetsov** [comment]
Why can't we mutate the `srv.TLSConfig`? Why are copying the lock in the first place?

**Comment by liggitt** [comment]
this is a copy of http.Server#ListenAndServeTLS with tweaks

**Pull request closed without merging**